pkgs/top-level/all-packages.nix

```diff
@@ -24557,6 +24557,10 @@ with pkgs;
 
   argocd = callPackage ../applications/networking/cluster/argocd { };
 
+  argocd-autopilot = callPackage ../applications/networking/cluster/argocd-autopilot {
+    buildGoModule = buildGo117Module;
```

Comment:
Isn't this the default already 🤔 ?

Reply:
Good catch! It is by now. Can be safely removed.

Reply:
I think we left it as is because of @sagikazarmark's comment, and points like:

- Go 1.17 is currently used by the application;
- Go 1.18 is being implemented in nixpkgs;
- No idea about the default for darwin;
- Maybe backport to 21.11 need this;

Reply:
AFAIK:

- We don't necessary use the same Go version as upstream unless there is a reason for it (e.g.: doesn't build with a newer version)
- The default for darwin should be the same
- On 21.11 backport you can set this